geopandas/io/tests/test_file.py

```diff
@@ -340,11 +340,44 @@ def test_read_file_filtered__rows_bbox__polygon(df_nybb):
     assert filtered_df_shape == (1, 5)
 
 
-def read_file_filtered_rows_invalid():
+def test_read_file_filtered_rows_invalid():
     with pytest.raises(TypeError):
         read_file(geopandas.datasets.get_path("nybb"), rows="not_a_slice")
 
 
+def test_read_file__columns():
+    gdf = geopandas.read_file(
+        geopandas.datasets.get_path("naturalearth_lowres"),
+        columns=["continent", "name"],
+    )
+    assert gdf.columns.tolist() == ["continent", "name", "geometry"]
+
+
+def test_read_file__columns_duplicate():
+    gdf = geopandas.read_file(
+        geopandas.datasets.get_path("naturalearth_lowres"),
+        columns=["continent", "name", "name"],
+    )
+    assert gdf.columns.tolist() == ["continent", "name", "name", "geometry"]
```

Comment:
@snowman2 thanks for addressing this!  Sorry for the confusion I caused.  What I meant was that the desired behavior should be to reduce to the unique ordered list of columns and ignore duplicates - _unlike_ the DataFrame constructor, but _like_ other interfaces that read from a file using a `columns` parameter (e.g., `read_parquet`).

So this should be 
```
assert gdf.columns.tolist() == ["continent", "name", "geometry"]
```

Reply:
Ha - I was thinking you wanted the `pandas` behavior. The logic I had previously used the order of the columns in the file and using the list of columns to filter them. Would that work?